packages/headless/src/composables/listbox/createListbox.ts

```diff
@@ -3,19 +3,29 @@ import { createId } from "../..";
 import { createBuilder, type HeadlessElementAttributes } from "../../utils/builder";
 import { useTypeAhead } from "../typeAhead";
 
-export type CreateListboxOptions<TValue extends ListboxValue> = {
+export type ListboxValue = string | number | boolean;
+
+export type ListboxModelValue<
+  TOption extends ListboxValue,
+  TMultiple extends boolean | undefined,
+> = TMultiple extends true ? TOption[] | undefined : TOption | undefined;
+
+export type CreateListboxOptions<
+  TMultiple extends boolean | undefined,
+  TOption extends ListboxValue = ListboxValue,
+> = {
```

Comment:
```suggestion
export type ListboxModelValue<
  TOption extends ListboxValue,
  TMultiple extends boolean,
> = TMultiple extends true ? TOption[] : TOption;

export type CreateListboxOptions<
  TMultiple extends boolean,
  TOption extends ListboxValue = ListboxValue,
> = {
```

Reply:
I think we have to keep the undefined in `TMultiple extends true ? TOption[] | undefined : TOption | undefined;` because a modelValue can always be undefined, right?

Reply:
aah it has a "?" in the types. allright :)